qmfkdldks/motion-menu-consumer
# Add Google Sign In button

# What is the objective?

Reduce the amount of cost necessary to start using the digital menu. Improve the onboarding process and automatically create the digital menu with great quality content.

# User story

As a visitor, I want to to click google sign in button so I can start using the QR menu.

# Acceptance criteria

- ~Visitors can see the google sign in button.~
- ~Visitors can click sign in button.~
- ~Google sign in event is tracked.~
- ~Sign in button should create an account.~
- Updat motion menu owners project.

# Technical Notes

Check official google sign in https://developers.google.com/identity/sign-in/web/build-button

# What should be tested

- Account should be created (Company record)
- Button should exist.
- Events

# QA Notes

1. Visit the landing page https://url.com/
2. Check if button exists.
3. Check if event is fired.
4. Check if it navigates to the profiling page.

# Engineering Pitfalls

-Troubleshoot problems with sveltekit packages:
`yarn upgrade @sveltejs/kit vite --latest`

-Read cookies on server side and use them in +page.ts 
```js

//+page.server.ts
export const load = async ({ cookies }) => {
  const cookies = cookies.get("cookie")

  //return cookies in +layout.server.ts to use in +layout.ts
  return {
    cookies: {
      cookie
    },
  };
};

//+page.ts
export const load = async ({ data: { cookies } }) => {
  const { cookie } = cookies;
...
}
```
